tfsdk/serve_test.go

```diff
@@ -1940,7 +2269,9 @@ func TestServerPlanResourceChange(t *testing.T) {
 				t.Errorf("Expected planned private to be %q, got %q", tc.expectedPlannedPrivate, got.PlannedPrivate)
 				return
 			}
-			if diff := cmp.Diff(got.RequiresReplace, tc.expectedRequiresReplace, cmpopts.EquateEmpty()); diff != "" {
+			if diff := cmp.Diff(got.RequiresReplace, tc.expectedRequiresReplace, cmpopts.EquateEmpty(), cmpopts.SortSlices(func(x, y *tftypes.AttributePath) bool {
```

Comment:
Hmm, rather than new modifiers on cmp.Diff, does implementing an `Equal` method allow us to make this work without the cmpopts?

Reply:
Yes, that would work. We could do one of the following:

1. Define a `type  RequiresReplacePaths []*tftypes.AttributePath` in the framework with an `Equal` method that ignores ordering
2. Do something similar but in plugin-go/tftypes itself.

However, I think it might be better just to have the framework server sort them so the response is deterministic, rather than ignoring the change of order during testing. Added 583f552d0530a5c7d4ddff4b551aa3da32eb8cd3 and a8991d73b60056734455d5afaded3611c1204fa0 for this purpose.